android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebChromeClient.java

```diff
@@ -97,22 +98,26 @@ public void onRequestPermissionsResult(
 						TiBaseActivity.unregisterPermissionRequestCallback(TiC.PERMISSION_CODE_LOCATION);
 
 						// Determine if location permission was granted.
-						boolean wasGranted = false;
+						boolean grantedFineLocation = false;
+						boolean grantedCoarseLocation = false;
 						if (permissions.length == grantResults.length) {
 							for (int index = 0; index < permissions.length; index++) {
 								if (Manifest.permission.ACCESS_FINE_LOCATION.equals(permissions[index])) {
-									wasGranted = (grantResults[index] == PackageManager.PERMISSION_GRANTED);
-									break;
+									grantedFineLocation = (grantResults[index] == PackageManager.PERMISSION_GRANTED);
+								}
+								if (Manifest.permission.ACCESS_COARSE_LOCATION.equals(permissions[index])) {
+									grantedCoarseLocation = (grantResults[index] == PackageManager.PERMISSION_GRANTED);
 								}
 							}
 						}
 
 						// Notify WebView whether or not location access was granted.
-						callback.invoke(origin, wasGranted, false);
+						callback.invoke(origin, grantedFineLocation && grantedCoarseLocation, false);
```

Comment:
This should be a bitwise-or `||`, not an `&&`.

That way it will match the `checkSelfPermission()` behavior you are doing above where you only need 1 of these permissions to pass.